### DISCUSSION TITLE
can it compile into wasm wasm ? at least some parts, like SQL parser + data mapper

### DISCUSSION THREAD

**Discussion by dzmitry-lahoda** [rbatis/rbatis#296] [General]
so I can plug connection and record mapping layers

**Comment by zhuxiujia** [comment]
> so I can plug connection and record mapping layers

Currently, Rbatis uses proc-Macro to convert HTML_SQL and Py_SQL into rust code during compilation, so the performance  Compared to Java's OGNL expression engine higher (cost is minimal or Zero cost abstraction)。

I don't understand  why are you    'plug connection and record mapping layers' using wasm

**Comment by dzmitry-lahoda** [comment]
wasm has wasi to access files. so for sure I cannot use any of file/network specific logic inside wasm. so i would like to use marco part and mapping some rust to sql parts, not actually running queries

**Comment by zhuxiujia** [comment]
can you elaborate on your problem?

**Comment by dzmitry-lahoda** [comment]
oh, i have sqlite compiled to wasm. it has its own connection type and execute method and custom return values etc.

i want to take some code from rbatis, and put it over that connection and mapping, how can I do that?
